src/transformer/descriptor/intersection/intersection.ts

```diff
@@ -9,7 +9,7 @@ export function GetIntersectionDescriptor(intersectionTypeNode: ts.IntersectionT
     const nodes: ts.Node[] = GetTypes(intersectionTypeNode.types, scope);
 
     const hasInvalidIntersections: boolean = nodes.some((node: ts.Node) => {
-        return TypescriptHelper.IsLiteralOrPrimitive(node);
+        return TypescriptHelper.IsLiteralOrPrimitive(node) || node.kind === ts.SyntaxKind.TypeQuery;
```

Comment:
I am trying to start using ts.isType instead of kind check... for example, ts.isTypeQuery(node)

Reply:
interesting, I've change it here, I think in busy places we should still use the check with kind to make it quicker